onnxruntime/core/mlas/lib/qgemm_kernel_amx.cpp

```diff
@@ -0,0 +1,730 @@
+/*++
+
+Copyright (c) Microsoft Corporation. All rights reserved.
+
+Licensed under the MIT License.
+
+Module Name:
+
+    qgemm_kernel_amx.cpp
+
+Abstract:
+
+    This module implements QGEMM kernels for amx.
+
+--*/
+
+#include "mlasi.h"
+#include "qgemm.h"
+
+
+#define TMM0 0
+#define TMM1 1
+#define TMM2 2
+#define TMM3 3
+#define TMM4 4
+#define TMM5 5
+#define TMM6 6
+#define TMM7 7
+
+#define KPACK (4 / sizeof(type_t))  // Vertical K packing into Dword
+
+#define TILE_M 16
+#define TILE_N 16
+#define TILE_K 64
+
+/*******************************************************************
+ * Packing and Gemm kernels for U8S8 AMX
+ ******************************************************************/
+struct MLAS_GEMM_U8S8_KERNEL_AMX {
+    typedef uint8_t PackedAType;
+    typedef uint8_t PackedBType;
+    typedef uint8_t OffsetAType;
+    typedef int8_t  OffsetBType;
+
+    static constexpr size_t PackedK = TILE_K;
+
+    // Use smaller stride for debugging,
+    static constexpr MLAS_GEMM_QUANT_STRIDES Strides{32, 128, 1024};
+    static constexpr MLAS_GEMM_QUANT_STRIDES PackedStrides{32, 512, 2048};
+};
+
+constexpr size_t MLAS_GEMM_U8S8_KERNEL_AMX::PackedK;
+constexpr MLAS_GEMM_QUANT_STRIDES MLAS_GEMM_U8S8_KERNEL_AMX::Strides;
```

Comment:
You may try this: https://www.codingame.com/playgrounds/2205/7-features-of-c17-that-will-simplify-your-code/inline-variables

Reply:
hmm, this pattern happens a lot in old MLAS. let me get another PR for this.